src/guides/migrationtoversion3.md

```diff
@@ -0,0 +1,137 @@
+---
+title: migration to nyxx_* 3.x
+author: l7ssha
+timestamp: 2021-12-17
+category: guides
+---
+
+`3.0.0` includes some big changes which are breaking to most of the code from previous version.
+This guide is going to be broken into parts for each library.
+
+Biggest change for each library are new entities model which incorporates interfaces for each class provided by library.
+Interfaces are now exposed instead of concrete implementations of class which allows easier extending, mocking and replacing nyxx
+and its components.
+
+## nyxx
+
+#### `INyxxFactory` and `connect` in INyxx
+
+  Since this release only way to spawn a new instance of nyxx is via `NyxxFactory` via `createNyxxWebsocket` which creates
```

Comment:
```suggestion
  Since this release the only way to spawn a new instance of nyxx is via `NyxxFactory`'s method `createNyxxWebsocket` which creates
```